src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -30,31 +34,44 @@
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_CREDITS, PREFIX_TAG);
+            ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_TAG);
 
-        Index index;
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty() || argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                EditCommand.MESSAGE_USAGE));
+        }
 
+        ModuleInfoSearcher moduleInfoSearcher = new ModuleInfoSearcher();
+        ModuleCode code = ParserUtil.parseModuleCode(argMultimap.getPreamble());
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            moduleInfoSearcher.searchModule(code.toString());
+        } catch (CommandException e) {
+            throw new ParseException(e.getMessage());
         }
 
         EditModuleDescriptor editModuleDescriptor = new EditModuleDescriptor();
+        ModuleInfo moduleInfo;
+
         if (argMultimap.getValue(PREFIX_CODE).isPresent()) {
-            editModuleDescriptor.setModuleCode(ParserUtil.parseModuleCode(argMultimap.getValue(PREFIX_CODE).get()));
-        }
-        if (argMultimap.getValue(PREFIX_CREDITS).isPresent()) {
-            editModuleDescriptor.setModularCredits(
-                    ParserUtil.parseModularCredits(argMultimap.getValue(PREFIX_CREDITS).get()));
+            ModuleCode newCode = ParserUtil.parseModuleCode(argMultimap.getValue(PREFIX_CODE).get());
+            try {
+                moduleInfo = moduleInfoSearcher.searchModule(newCode.toString());
+                editModuleDescriptor.setModuleCode(newCode);
+                editModuleDescriptor.setModuleTitle(ParserUtil.parseModuleTitle(moduleInfo.getTitle()));
+                editModuleDescriptor.setModularCredits(
+                    ParserUtil.parseModularCredits(moduleInfo.getModuleCredit()));
+            } catch (CommandException e) {
+                throw new ParseException(e.getMessage());
+            }
```

Comment:
Currently, the check for the new module and the module to be edited both throws the same error message "Failed to find module", if it is an invalid module. Hence, if a user does "edit CS1111 c/CS2222" (two wrong modules), they wouldnt be able to tell which module is invalid.

Maybe we can consider having personalised error messages for the two different module checks, so that the user will know which module is invalid.

Reply:
Ah yes bro i actually did think about this!! If you see the EditCommandParser class you can see 2 messages that i didn't use hahahaha nice catch though👍🏽

Anyway i think i will open up a new issue to clean up all the messages in general